src/core_inner/bob_client.rs

```diff
@@ -153,6 +153,33 @@ pub(crate) mod b_client {
                 result: BackendError::Timeout,
             })
         }
+
+        #[allow(dead_code)]
+        pub fn exist(&mut self, keys: Vec<BobKey>, options: GetOptions) -> Exist {
+            let mut client = self.client.clone();
+            let node1 = self.node.clone();
+            let node2 = self.node.clone();
+            Exist(Box::pin(async move {
+                client
+                    .exist(Request::new(ExistRequest {
+                        keys: keys.into_iter().map(|k| BlobKey { key: k.key }).collect(),
+                        options: Some(options),
+                    }))
+                    .map(move |r| {
+                        r.map(|r| ClusterResult {
```

Comment:
you can create ClusterResult after client.exists(..). And you wont have to clone node twice